### PR TITLE
grpc: Address CVE-2023-32067 

### PR DESCRIPTION
**Pull request by Ankita13-code**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
This PR addresses CVE-2023-32067 for grpc submodules package.

###### Change Log  

- Add formatted patch for this CVE in grpc
- Upstream patch - [CVE-2023-32067 patch](https://github.com/c-ares/c-ares/commit/b9b8413cfdb70a3f99e1573333b23052d57ec1ae)

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-32067

###### Test Methodology

- Pipeline build id: [buddy build](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=691909&view=results)

### CODE REVIEW

File: SPECS/grpc/grpc.spec [hunk 1]
```diff
@@ -69,6 +71,7 @@ Python language bindings for gRPC.
 %prep
 %setup -q -n %{name}-%{version}
 %setup -T -D -a 1
+%patch0 -p1
```
Comment (jslobodzian) [inline SPECS/grpc/grpc.spec:74]: @Ankita13-code prefer updating to %autopatch -p1 in this situation.

Comment (rikenm1) [inline SPECS/grpc/grpc.spec:74]:
```suggestion
%autopatch -p1
```

**Review by rikenm1 — CHANGES_REQUESTED** [review]

**Review by rikenm1 — APPROVED** [review]
LGTM

**Review by jslobodzian — APPROVED** [review]

**Comment by nicogbg** [comment]
Auto cherry-pick results:
- main :white_check_mark: -> https://github.com/microsoft/azurelinux/pull/11605

Auto cherry-pick pipeline run -> https://dev.azure.com/mariner-org/mariner/_build/results?buildId=697908&view=results

**Pull request merged**